skyportal/handlers/api/observation_plan.py

```diff
@@ -1720,120 +1728,147 @@ async def get(self, observation_plan_request_id):
         )
 
         group_ids = self.get_query_argument('group_ids', None)
-        if not group_ids:
-            groups = self.current_user.accessible_groups
-        else:
-            try:
-                groups = Group.get_if_accessible_by(
-                    group_ids, self.current_user, raise_if_none=True
-                )
-            except AccessError:
-                return self.error('Could not find any accessible groups.', status=403)
 
-        options = [
-            joinedload(ObservationPlanRequest.observation_plans)
-            .joinedload(EventObservationPlan.planned_observations)
-            .joinedload(PlannedObservation.field)
-        ]
+        with DBSession() as session:
 
-        observation_plan_request = ObservationPlanRequest.get_if_accessible_by(
-            observation_plan_request_id,
-            self.current_user,
-            mode="read",
-            raise_if_none=True,
-            options=options,
-        )
-        self.verify_and_commit()
+            if not group_ids:
+                groups = self.current_user.accessible_groups
+            else:
+                try:
+                    groups = Group.get_if_accessible_by(
+                        group_ids, self.current_user, raise_if_none=True
+                    )
+                except AccessError:
+                    return self.error(
+                        'Could not find any accessible groups.', status=403
+                    )
 
-        allocation = Allocation.get_if_accessible_by(
-            observation_plan_request.allocation_id,
-            self.current_user,
-            raise_if_none=True,
-        )
+            options = [
+                joinedload(ObservationPlanRequest.observation_plans)
+                .joinedload(EventObservationPlan.planned_observations)
+                .joinedload(PlannedObservation.field)
+            ]
 
-        localization = (
-            Localization.query_records_accessible_by(
+            observation_plan_request = ObservationPlanRequest.get_if_accessible_by(
+                observation_plan_request_id,
                 self.current_user,
+                mode="read",
+                raise_if_none=True,
+                options=options,
             )
```

Comment:
There's a new `get` method in baselayer that uses the new select tools. And also a `select` method that does permissioning. Since you are already in a session might as well use that.

Reply:
Great.